src/test/scala/org/constellation/domain/transaction/TransactionChainServiceTest.scala

```diff
@@ -37,17 +39,145 @@ class TransactionChainServiceTest
       last.unsafeRunSync shouldBe LastTransactionRef("", 0L)
     }
 
-    "should return last transaction ref if address is known" in {}
+    "should return last transaction ref if address is known" in {
+      val address = "DAG123"
+      val lastTx = LastTransactionRef("tx123", 1)
+      service.lastTransactionRef.set(Map(address -> lastTx)).unsafeRunSync
+
+      val last = service.getLastTransactionRef(address).unsafeRunSync
+      last shouldEqual lastTx
+    }
+  }
+
+  "getLastAcceptedTransactionRef" - {
+    "should return empty string and 0 ordinal if address is unknown" in {
+      val lastAccepted = service.getLastAcceptedTransactionRef("unknown").unsafeRunSync
+
+      lastAccepted shouldEqual LastTransactionRef("", 0L)
+    }
+
+    "should return last accepted transaction ref if address is known" in {
+      val address = "DAG123"
+      val lastTx = LastTransactionRef("tx123", 1)
+      service.lastAcceptedTransactionRef.set(Map(address -> lastTx)).unsafeRunSync
+      val lastAccepted = service.getLastAcceptedTransactionRef(address).unsafeRunSync
+
+      lastAccepted shouldEqual lastTx
+    }
+  }
+
+  "getLastAcceptedTransactionMap" - {
+    "should return current map of accepted transactions" in {
+      service.getLastAcceptedTransactionMap().unsafeRunSync shouldEqual Map.empty
```

Comment:
I see we're using wrong name here. it's not a map of accepted txs but of accepted tx refs. Do you mind changing the name of fn to `getLastAcceptedTransactionRefMap`?